test/runtests.jl

```diff
@@ -1,128 +1,153 @@
 
-    @safetestset "InputOutput - hdf5" begin @time include("InputOutput/hdf5.jl") end
-    @safetestset "InputOutput - spectralelement2d" begin @time include("InputOutput/spectralelement2d.jl") end
-    @safetestset "InputOutput - hybrid2dbox" begin @time include("InputOutput/hybrid2dbox.jl") end
-    @safetestset "InputOutput - hybrid2dbox_topography" begin @time include("InputOutput/hybrid2dbox_topography.jl") end
-    @safetestset "InputOutput - hybrid2dbox_stretched" begin @time include("InputOutput/hybrid2dbox_stretched.jl") end
-    @safetestset "InputOutput - hybrid3dbox" begin @time include("InputOutput/hybrid3dbox.jl") end
-    @safetestset "InputOutput - hybrid3dcubedsphere" begin @time include("InputOutput/hybrid3dcubedsphere.jl") end
-    @safetestset "InputOutput - hybrid3dcubedsphere_topography" begin @time include("InputOutput/hybrid3dcubedsphere_topography.jl") end
+tabulate_tests(unit_tests; include_times = false)
 
-    @safetestset "Array interpolation" begin @time include("Remapping/interpolate_array.jl") end
-    @safetestset "Array interpolation" begin @time include("Remapping/distributed_remapping.jl") end
+nameify(name) = replace(name, "/" => "", ".jl" => "", " " => "")
+# Hashes can result in `cannot assign a value to imported variable Main.include` error.
+gensym_no_hashes(x) = replace(string(gensym(x)), "#" => "")
+modulename(name) = gensym_no_hashes(nameify(name))
+moduledef(test) =
+    "module $(modulename(test.filename))\ninclude(\"$(test.filename)\")\nend"
 
-    # Code quality checks
-    @safetestset "Aqua" begin @time include("aqua.jl") end
-end
 import ClimaComms
-if ClimaComms.device() isa ClimaComms.CUDADevice
-    @safetestset "GPU - cuda" begin @time include("gpu/cuda.jl") end
-    @safetestset "GPU - data" begin @time include("DataLayouts/cuda.jl") end
-    @safetestset "GPU - spaces" begin @time include("Spaces/spaces.jl") end
-    @safetestset "Spaces - serial CUDA DSS" begin @time include("Spaces/ddss1.jl") end
-    @safetestset "Spaces - serial CUDA DSS on CubedSphere" begin @time include("Spaces/ddss1_cs.jl") end
-    @safetestset "Operators - spectral element CUDA" begin @time include("Operators/spectralelement/rectilinear_cuda.jl") end
-    @safetestset "Operators - finite difference CUDA" begin @time include("Operators/hybrid/unit_cuda.jl") end
-    @safetestset "Operators - extruded sphere space operators CUDA" begin @time include("Operators/hybrid/extruded_sphere_cuda.jl") end
-    @safetestset "Operators - extruded 3dbox space operators CUDA" begin @time include("Operators/hybrid/extruded_3dbox_cuda.jl") end
-    @safetestset "Fields - CUDA mapreduce" begin @time include("Fields/reduction_cuda.jl") end
+prevent_leaky_tests = false
+# for test in [unit_tests[1], unit_tests[2]] # for debugging this file
+for test in unit_tests
+    if test.gpu_only && !(ClimaComms.device() isa ClimaComms.CUDADevice)
+        continue
+    end
+    @info "--- About to test $(test.filename)"
```

Comment:
```suggestion
    @info "About to test $(test.filename)"
```
Are the dashes used for anything?

Reply:
I'll remove it, I think it's actually a bit too noisy

Reply:
I'm actually going to leave them commented, just for debugging